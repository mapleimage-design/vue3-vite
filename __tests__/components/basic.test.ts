import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import Hello from "../../src/components/HelloWorld.vue";



describe("import vue components", () => {
  test("mounted component", async () => {
    expect(Hello).toBeTruthy();
    const wrapper = mount(Hello, {
      props: {
        msg: "123",
      },
    });
    expect(wrapper.text()).toBe("1230click");
    expect(wrapper.html()).matchSnapshot();
    await wrapper.get("button").trigger("click");
    expect(wrapper.text()).toBe("1231click");
    await wrapper.get("button").trigger("click");
    await wrapper.get("button").trigger("click");
    expect(wrapper.text()).toBe("1233click");
  });
});
