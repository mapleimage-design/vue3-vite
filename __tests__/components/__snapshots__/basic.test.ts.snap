// Vitest Snapshot v1

exports[`import vue components > mounted component 1`] = `
"<h1 data-v-469af010=\\"\\">123</h1>
<h1 data-v-469af010=\\"\\">0</h1>
<button data-v-469af010=\\"\\">click</button>"
`;
